crates/cargo-util-schemas/src/manifest/mod.rs

```diff
@@ -776,12 +776,22 @@ impl<'de, P: Deserialize<'de> + Clone> de::Deserialize<'de> for TomlDependency<P
     where
         D: de::Deserializer<'de>,
     {
+        use serde::de::Error as _;
+        let expected = "a version string like \"0.9.8\" or a \
+                     detailed dependency like { version = \"0.9.8\" }";
         UntaggedEnumVisitor::new()
-            .expecting(
-                "a version string like \"0.9.8\" or a \
-                     detailed dependency like { version = \"0.9.8\" }",
-            )
+            .expecting(expected)
             .string(|value| Ok(TomlDependency::Simple(value.to_owned())))
+            .bool(|value| {
+                let err = format!(
+                    "invalid type: boolean `{value}`, expected {expected}\n\
+                    note: if you meant to use a workspace member, you can write\n \
+                      dep = {{ workspace = {value} }}\n\
+                    or\n \
+                      dep.workspace = {value}",
```

Comment:
This will provide a wrong message for `dep = false` (unsupported),

Should we skip the hint in that case?

Reply:
make sense we can show the default(`expected`) hint for `false` bool...this is what you mean rigiht?

Reply:
Yup

Reply:
@epage I updated my previous comment lmk your thought please else I will continue with showing `expected` message for `false` value.

Reply:
nvm actually, I thought about it for a sec again and I think your suggestion is the obvious choice